networkx/algorithms/dag.py

```diff
@@ -287,3 +289,75 @@ def is_aperiodic(G):
         return g==1
     else:
         return g==1 and nx.is_aperiodic(G.subgraph(set(G)-set(levels)))
+
+def transitive_closure(G):
+    """ Based on http://www.ics.uci.edu/~eppstein/PADS/PartialOrder.py
+
+    The transitive closure of G = (V,E) is a graph G+ = (V,E+) such that
+    for all v,w in V there is an edge (v,w) in E+ if and only if there
+    is a non-null path from v to w in G.
+
+    Parameters
+    ----------
+    G : NetworkX DiGraph
+      Graph
+
+    Returns
+    -------
+    TC : NetworkX DiGraph
+       Graph
+
+    Raises
+    ------
+    NetworkXError
+      If G is not directed
+
+    """
+    if not G.is_directed():
+        raise nx.NetworkXError("transitive_closure not defined for undirected graphs")
```

Comment:
That's correct. I was suggesting to use the `not_implemented_for` decorator instead of an if clause in the function body to check that the input Graph is directed. But this is a very minor thing, don't worry about this.